README.md

```diff
@@ -651,9 +651,84 @@ Filtering just American and European countries, leaving out countries whose main
 - run: sort
   parameters:
     resources: world_population
-    sort-by: "{country_name}" 
+    sort-by: "{country_name}"
 ```
 
+### ***`add_fields`***
+
+Add field(s) to streamed resources
+
+`add_fields` accepts a list of resources and fields to add to existing resource. It will output the rows for each resource with new field(s) (columns) in it. `add_fields` allows to perform various operations before inserting value into targeted field.
+
+_Parameters_:
+
+- `resources` - Which resources to sort. Same semantics as `resources` in `stream_remote_resources`.
+- `fields` - List of the fields with their metadata to be added to resource
+  - `operation`: operation to perform on values of pre-defined columns of the same row. available operation:
+    - `constant` - add a constant value
+    - `sum` - summed value for given columns in a row.
+    - `avg` - average value from given columns in a row.
+    - `min` - minimum value among given columns in a row.
+    - `max` - maximum value among given columns in a row.
+    - `multiply` - product of given columns in a row.
+    - `join` - joins two or more column values in a row (separated by comma)
+    - `format` - Python format string used to form the value Eg:  `my name is {first_name}`
+  - `target` - name of the new field
+  - `nullas` - value if column is empty (default: 0)
+  - `type` - valid [table-shcema](https://frictionlessdata.io/specs/table-schema/#types-and-formats) type (default: `any`)
+  - `columns` - list of columns the operations should be performed on (Not required in case of `format` and `constant`)
+  - `formated-string` - Python format string with exiting column values Eg: `{first_name} {last_name}` (operation should be set to `format`)
+  - `constant` - constant value for field (operation should be set to `constant`)
```

Comment:
instead of `constant`, `formated-string`, let's have a single parameter `with`:
- used for constant value when constant operation
- used for delimiter for join
- user for format string for format